compiler/transf.nim

```diff
@@ -364,7 +364,7 @@ proc transformYield(c: PTransf, n: PNode): PNode =
   if e.typ.isNil: return result # can happen in nimsuggest for unknown reasons
   if c.transCon.forStmt.len != 3:
     e = skipConv(e)
-    if e.kind in {nkPar, nkTupleConstr}:
```

Comment:
Ref https://github.com/nim-lang/Nim/pull/13793

tuple expr is already unified, so I removed `nkPar`.